files/d7_sync.sh

```diff
@@ -2,64 +2,69 @@
 ## Sync Drupal files & DB from source host
 PATH=/opt/d7/bin:/usr/local/bin:/usr/bin:/bin:/sbin:$PATH
 
-# Writable dir on both local and souce hosts
-TEMPDIR=/var/local/backups/drupal/temp
+source /opt/d7/etc/d7-conf.sh
 
 ## Require arguments
-if [ ! -z "$1" ] && [ ! -z "$2" ]
+if [ ! -z "$1" ] && [ ! -z "$2" ] && [ ! -z "$3" ]
 then
-  SITEPATH=$1
+  ORIGIN_SITEPATH=$1
   SRCHOST=$2
-  echo "Syncing $SITEPATH content from $SRCHOST"
+  SITEPATH=$3
+  echo "Syncing $ORIGIN_SITEPATH content from $SRCHOST to local $SITEPATH"
 else
-  echo "Requires site path (eg. /srv/sample) and source host as argument"
+  echo "Requires site path (eg. /srv/sample), source host, and new site path as arguments"
   exit 1;
 fi
 
+## Grab the basename of the NEW site to use in a few places.
+SITE=$(basename "$SITEPATH")
+
+
 ## Init site if it doesn't exist
 if [[ ! -e $SITEPATH ]]; then
-    sudo d7_init.sh $SITEPATH || exit 1;
+    sudo d7_init.sh "$SITEPATH" || exit 1;
 fi
 
-## Grab the basename of the site to use in a few places.
-SITE=`basename $SITEPATH`
-
 ## Make the sync directory
-sudo mkdir -p $SITEPATH/default/files_sync
-sudo chmod 777 $SITEPATH/default/files_sync
+sudo mkdir -v  -p "$SITEPATH/default/files_sync"
+sudo chmod 777 "$SITEPATH/default/files_sync"
 
-## Sync Files to sync directory
-rsync -a --ignore-times --omit-dir-times --no-perms $SRCHOST:$SITEPATH/default/files/ $SITEPATH/default/files_sync || exit 1;
+## Sync Files to writable directory (sudo would break ssh)
+RSOPTS="--verbose --recursive --links --devices --compress"
+rsync  $RSOPTS  "$SRCHOST:$ORIGIN_SITEPATH/default/files/" "$SITEPATH/default/files_sync" ;
 echo "Files synced."
 
 ## Set perms for sync directory
 echo "Setting permissions for synced files."
-sudo find $SITEPATH/default/files_sync -type d -exec chmod u=rwx,g=rx,o= '{}' \;
-sudo find $SITEPATH/default/files_sync -type f -exec chmod u=rw,g=r,o= '{}' \;
-sudo chown -R apache:apache $SITEPATH/default/files_sync
+sudo find "$SITEPATH/default/files_sync" -type d -exec chmod u=rwx,g=rx,o= '{}' \;
+sudo find "$SITEPATH/default/files_sync" -type f -exec chmod u=rw,g=r,o= '{}' \;
+sudo chown -R apache:apache "$SITEPATH/default/files_sync"
 echo "Setting SELinux for synced files."
 sudo semanage fcontext -a -t httpd_sys_rw_content_t  "$SITEPATH/default/files_sync(/.*)?" || exit 1
-sudo restorecon -R $SITEPATH/default || exit 1;
+sudo restorecon -R "$SITEPATH/default" || exit 1;
 
 ## Now that everything is ready, swap in the synced files
 echo "Placing synced files."
-sudo rm -rf $SITEPATH/default/files_bak
-mv $SITEPATH/default/files $SITEPATH/default/files_bak
```

Comment:
The things not done as sudo should work for users in the apache group.  If the operation is failing, we probably need to check group memberships or permissions instead of running as root.